server/src/main/java/org/elasticsearch/telemetry/metric/Meter.java

```diff
@@ -62,6 +67,20 @@ public interface Meter {
      */
     DoubleGauge getDoubleGauge(String name);
 
+    DoubleGaugeObserver registerDoubleGaugeObserver(String name, String description, String unit, DoubleSupplier observed);
+
+    DoubleGaugeObserver registerDoubleGaugeObserver(
+        String name,
+        String description,
+        String unit,
+        DoubleSupplier observed,
+        Map<String, Object> attributes
```

Comment:
do we want to have attributes set only once? or should they also be 'observed'. so that if attributes changes they would be exported

Reply:
There are two flavors of this API.  One does have them [observed](https://github.com/elastic/elasticsearch/pull/100644/files#diff-41bf6f2bbc6779e5cb9549ab5fb63e4fa8e5cd1286de40bc6dbeb0f87a37b88aR58-R64) along with the value.